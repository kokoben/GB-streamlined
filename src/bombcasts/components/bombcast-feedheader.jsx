import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import FeedHeader from '../../components/feed-header';
import { fetchBombcastSearchVideos, setBombcastVideo } from '../actions/index';
import BombcastSearchResultsSelector from '../selectors/current-bombcast-search-results';

const BombcastFeedHeader = props => (
  <FeedHeader
    category="Bombcast"
    placeholder="Search for Bombcast videos..."
    fetchVideos={props.fetchBombcastSearchVideos}
    currentResults={props.currentBombcastSearchResults}
    setVideo={props.setBombcastVideo}
  />
);

/* eslint-disable react/forbid-prop-types */
BombcastFeedHeader.propTypes = {
  fetchBombcastSearchVideos: PropTypes.func.isRequired,
  currentBombcastSearchResults: PropTypes.array.isRequired,
  setBombcastVideo: PropTypes.func.isRequired,
};
/* eslint-enable */

const mapStateToProps = state => ({
  currentBombcastSearchResults: BombcastSearchResultsSelector(state),
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    fetchBombcastSearchVideos,
    setBombcastVideo,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(BombcastFeedHeader);
