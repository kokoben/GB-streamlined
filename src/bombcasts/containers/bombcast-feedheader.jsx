import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import FeedHeader from '../../components/feed-header';
import { fetchBombcastSearchVideos, setBombcastVideo } from '../actions/index';
import BombcastSearchResultsSelector from '../selectors/current-bombcast-search-results';

class BombcastFeedHeader extends Component {
  render() {
    console.log('current feature search reuslts: ', this.props.currentBombcastSearchResults);
    return (
      <FeedHeader
        category="Bombcast"
        placeholder="Search for Bombcast videos..."
        fetchVideos={this.props.fetchBombcastSearchVideos}
        currentResults={this.props.currentBombcastSearchResults}
        setVideo={this.props.setBombcastVideo}
      />
    );
  }
}

/* eslint-disable react/forbid-prop-types */
BombcastFeedHeader.propTypes = {
  fetchBombcastSearchVideos: PropTypes.func.isRequired,
  currentBombcastSearchResults: PropTypes.array.isRequired,
  setBombcastVideo: PropTypes.func.isRequired,
};
/* eslint-enable */

const mapStateToProps = state => ({
  currentBombcastSearchResults: BombcastSearchResultsSelector(state),
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    fetchBombcastSearchVideos,
    setBombcastVideo,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(BombcastFeedHeader);
