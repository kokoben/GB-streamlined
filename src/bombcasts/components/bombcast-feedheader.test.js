import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BombcastFeedHeader } from './bombcast-feedheader';

describe('bombcast-feedheader tests', () => {
  it('renders correctly', () => {
    const mockFetchBombcastSearchVideos = jest.fn();
    const mockSetBombcastVideo = jest.fn();
    const mockSetVideo = jest.fn();

    const wrapper = shallow(<BombcastFeedHeader
      currentBombcastSearchResults={[]}
      currentResults={[]}
      fetchBombcastSearchVideos={mockFetchBombcastSearchVideos}
      setBombcastVideo={mockSetBombcastVideo}
      setVideo={mockSetVideo}
    />);

    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});

