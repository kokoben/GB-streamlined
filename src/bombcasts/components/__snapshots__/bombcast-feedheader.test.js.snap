// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bombcast-feedheader tests renders correctly 1`] = `
<FeedHeader
  category="Bombcast"
  currentResults={Array []}
  fetchVideos={[Function]}
  placeholder="Search for Bombcast videos..."
  setVideo={[Function]}
/>
`;
